include/fe/fe_abstract.h

```diff
@@ -244,35 +244,68 @@ class FEAbstract : public ReferenceCountedObject<FEAbstract>
   /**
    * \returns The \p xyz spatial locations of the quadrature
    * points on the element.
+   *
+   * It is overwritten by infinite elements since there
+   * \p FEMap cannot be used to compute \p xyz.
    */
+#ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
+  virtual
+#endif
   const std::vector<Point> & get_xyz() const
   { calculate_map = true; return this->_fe_map->get_xyz(); }
 
+
+#ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
+  /**
+   * This function is the variant of \p get_JxW() for \p InfFE.
+   * Since J diverges there, a respectize decay-function must be
+   * applied to obtain well-defined quantities.
+   *
+   * For FE, it is equivalent to the common \p get_JxW().
+   */
+  virtual const std::vector<Real> & get_JxWxdecay_sq () const
```

Comment:
Is it true that someone using `InfFEs` should *only* use the "decay" JxW or do they need to have access to the non-decay JxW as well? As for a name I would prefer `JxW_decay` or `decay_JxW` to this one.

Reply:
> I would prefer JxW_decay or decay_JxW to this one.

You are right, this looks much cleaner (I especially like the idea of adding ``decay`` in front), but I wanted the ``sq`` because the ``decay``-factor here is the square of that which is compensated for in the shape functions.

> Is it true that someone using InfFEs should only use the "decay" JxW 

yes: JxW is a function that (when written in physical space) which grows like r^4; this behaviour is the main reason for all these changes:
  * Before, it was computed via Lagrange-interpolation on the reference-element, where the local coordinate ``zeta = 1-2a/r``, i.e. JxW goes like ``1/zeta^4``: being non-polynomial, the interpolation won't work properly
  * proper evaluation of the integral **must** yield a divergent ``JxW`` (because the volume is infinitely large)
  * the integral over the full matrix elements does not diverge because the shape functions decay fast enough. So I introduced some scaling in ``JxW`` which is compensated in ``phi``, avoiding divergences and still having the correct matrix elements.

**I'll try to get the 'documentation' mentioned above into shape asap; this might help to grasp the ideas behind**, than it probably is easier to discuss the naming-schemes and you have a chance to follow my changes. _I think I should have started with sharing the document and not purely the code; sorry for that_